src/components/NftRewards/NftRewardsSection.tsx

```diff
@@ -91,10 +107,16 @@ export function NftRewardsSection() {
     validatePayAmount?.(newPayAmount)
   }
 
-  const handleTierSelect = (tierId: number | undefined) => {
+  const handleTierSelect = (
+    tierId: number | undefined,
+    quantity: number, // quantity to select
+  ) => {
     if (!tierId || !rewardTiers) return
 
-    const newSelectedTierIds = [...(payMetadata?.tierIdsToMint ?? []), tierId]
+    const newSelectedTierIds = [...(payMetadata?.tierIdsToMint ?? [])].concat(
```

Comment:
**suggestion** there's no need to do spreads ... if we're not mutating any data.